.github/workflows/ci.yml

```diff
@@ -31,9 +31,9 @@ jobs:
               uses: actions/cache@v2
               with:
                   path: vendor
-                  key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+                  key: ${{ runner.os }}-php-${{ matrix.php-versions }}
```

Comment:
👍

Reply:
Maybe it would be better to just remove the caching here? Caching it per PHP version isn't really caching since the cache will only apply to one version.

Reply:
I thought this resulted in 4 different keys and 4 different associated caches. These 4 distinct cached sets of dependencies are available between multiple runs of the Github Actions. So instead of having a single cached set of dependencies there are now 4 sets of dependencies cached.

But I think I will have to add the `hashFiles` again to make sure the cache is invalidated when the `composer.lock` file changes.

Reply:
Ah I didn't know the cache persisted between multiple runs.
But yeah like you said, now we'd cache it indefinitely and never do a fresh install since there's no changes that can invalidate the cache.
Problem is that you also cannot name the cache based on a hash of the `composer.lock` file, because it doesn't exist yet at this point I think? So there's nothing to hash?
The `composer.lock` only gets created in the next step, when running `composer install`.
The reason it's not included in this repo is because this is a package, not an app, and when we install this package in one or more of our apps composer ignores the (nested) composer.lock anyway. So it creates a false sense of stability.

Reply:
I checked the Composer docs and they mention that we _can_ include it so we can test against the same version, but it will indeed be ignored when it installs this package in another project:

> For your library you may commit the composer.lock file if you want to. This can help your team to always test against the same dependency versions. However, this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.

https://getcomposer.org/doc/02-libraries.md#lock-file

This in contrast with their advice to commit the lock file for other projects: https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control

Reply:
We could also create the `hash` from the `composer.json` file?

Reply:
Yes, but then dependencies will still not update in the cache when a newer version is available, only when we change/add a dependency in `composer.json` ourselves

Reply:
True! So best to add a `composer.lock` file or skip the Github Actions caching for now?

Reply:
Yes, I'd personally go for the latter because then our approach stays consistent with other packages we have (which don't have a lock file), plus we would always run our tests against the latest dependency versions that match our constraints in composer.json, which are also the ones that would be installed in a parent project.